src/grammar.rs

```diff
@@ -104,7 +104,7 @@ impl Grammar {
 
         let mut result = String::new();
         for term in expression.terms_iter() {
-            match self.eval_terminal(&term) {
+            match self.eval_terminal(&term, rng) {
```

Comment:
Does this mean each term gets its own copy of the `StdRng` ? Seems strange for each term to essentially be given the same seed, or does it do something else?